File: apps/react-native/src/screens/HelpSupportScreen.js
import { ScrollView, StyleSheet, Text, TouchableOpacity, View } from 'react-native';
import { useSafeAreaInsets } from 'react-native-safe-area-context';
import { useHandedness, useTheme } from '../contexts/AppProviders';

export default function HelpSupportScreen({ navigation }) {
  const insets = useSafeAreaInsets();
  const { isLeftHanded } = useHandedness();
  const { colors } = useTheme();

  return (
    <View style={[styles.container, { backgroundColor: colors.background }]}>
      {/* Header */}
      <View
        style={[
          styles.header,
          { paddingTop: insets.top + 12 },
          isLeftHanded && styles.headerReversed,
        ]}
      >
        <TouchableOpacity
          style={styles.backButton}
          onPress={() => navigation.goBack()}
          accessibilityLabel="Go back"
        >
          <Text style={[styles.backIcon, { color: colors.text }]}>←</Text>
        </TouchableOpacity>

        <Text style={[styles.title, { color: colors.text }]}>Help / Support</Text>
      </View>

      <ScrollView
        contentContainerStyle={[
          styles.content,
          { paddingBottom: insets.bottom + 40 },
        ]}
      >
        <Text style={[styles.p, { color: colors.textSecondary }]}>
          If you need help using CareConnect, review the guidance below.
        </Text>

        <Text style={[styles.h2, { color: colors.text }]}>Common Topics</Text>
        <Text style={[styles.p, { color: colors.textSecondary }]}>
          • How to switch handedness layout{'\n'}
          • How to enable accessibility modes{'\n'}
          • How to view tasks and messages{'\n'}
          • How to change notification settings
        </Text>

        <Text style={[styles.h2, { color: colors.text }]}>Accessibility</Text>
        <Text style={[styles.p, { color: colors.textSecondary }]}>
          CareConnect includes UI-only accessibility profiles (Low Vision, Tremor/Motor, STML,
          Hearing Impaired). Functional mitigations may be implemented later.
        </Text>

        <Text style={[styles.h2, { color: colors.text }]}>Contact</Text>
        <Text style={[styles.p, { color: colors.textSecondary }]}>
          Support is not active in this prototype. For course support, contact your instructor.
        </Text>

        <Text style={[styles.note, { color: colors.textSecondary }]}>
          Tip: In production apps, this screen usually includes email support, FAQs, and a
          “Report a problem” flow.
        </Text>
      </ScrollView>
    </View>
  );
}

const styles = StyleSheet.create({
  container: { flex: 1 },

  header: {
    flexDirection: 'row',
    alignItems: 'center',
    paddingHorizontal: 20,
    paddingBottom: 12,
  },
  headerReversed: {
    flexDirection: 'row-reverse',
  },
  backButton: {
    padding: 8,
    marginRight: 12,
  },
  backIcon: {
    fontSize: 24,
  },
  title: {
    fontSize: 22,
    fontWeight: '800',
  },

  content: {
    paddingHorizontal: 20,
    paddingTop: 12,
  },

  h2: {
    fontSize: 16,
    fontWeight: '800',
    marginTop: 18,
    marginBottom: 6,
  },
  p: {
    fontSize: 14,
    lineHeight: 20,
  },
  note: {
    marginTop: 24,
    fontSize: 13,
    lineHeight: 18,
    fontStyle: 'italic',
  },
});

File: apps/flutter/lib/features/help/help_support_screen.dart
// lib/features/help/help_support_screen.dart

import 'package:flutter/material.dart';
import 'package:provider/provider.dart';

import '../../core/tokens/spacing.dart';
import '../../core/accessibility/app_settings_controller.dart';

class HelpSupportScreen extends StatelessWidget {
  const HelpSupportScreen({super.key});

  @override
  Widget build(BuildContext context) {
    final settings = context.watch<AppSettingsController>();
    final isLeftAligned = settings.isLeftAligned;
    final text = Theme.of(context).textTheme;

    final back = IconButton(
      tooltip: 'Go back',
      icon: const Icon(Icons.arrow_back),
      onPressed: () => Navigator.of(context).maybePop(),
    );

    return Scaffold(
      backgroundColor: const Color(0xFFF7FAFB),
      appBar: AppBar(
        title: const Text('Help / Support'),
        leading: isLeftAligned ? null : back,
        actions: isLeftAligned ? [back] : null,
      ),
      body: SafeArea(
        child: SingleChildScrollView(
          padding: const EdgeInsets.all(AppSpacing.lg),
          child: Column(
            crossAxisAlignment: CrossAxisAlignment.start,
            children: [
              Text(
                'If you need help using CareConnect, review the guidance below.',
                style: text.bodyMedium?.copyWith(color: Colors.grey.shade700),
              ),

              const SizedBox(height: 18),
              _H2('Common Topics'),
              _P(
                '• How to switch handedness layout\n'
                '• How to enable accessibility modes\n'
                '• How to view tasks and messages\n'
                '• How to change notification settings',
              ),

              _H2('Accessibility'),
              _P(
                'CareConnect includes UI-only accessibility profiles (Low Vision, Tremor/Motor, STML, Hearing Impaired). '
                'Functional mitigations may be implemented later.',
              ),

              _H2('Contact'),
              _P('Support is not active in this prototype. For course support, contact your instructor.'),

              const SizedBox(height: 22),
              Text(
                'Tip: In production apps, this screen usually includes email support, FAQs, and a “Report a problem” flow.',
                style: text.bodySmall?.copyWith(
                  fontStyle: FontStyle.italic,
                  color: Colors.grey.shade600,
                ),
              ),
            ],
          ),
        ),
      ),
    );
  }
}

class _H2 extends StatelessWidget {
  final String s;
  const _H2(this.s);

  @override
  Widget build(BuildContext context) {
    return Padding(
      padding: const EdgeInsets.only(top: 14, bottom: 6),
      child: Text(
        s,
        style: const TextStyle(fontSize: 16, fontWeight: FontWeight.w800),
      ),
    );
  }
}

class _P extends StatelessWidget {
  final String s;
  const _P(this.s);

  @override
  Widget build(BuildContext context) {
    return Text(
      s,
      style: const TextStyle(fontSize: 14, height: 1.45),
    );
  }
}
